sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisTest.scala

```diff
@@ -200,11 +200,15 @@ trait AnalysisTest extends PlanTest {
     }
   }
 
-  protected def interceptParseException(
-      parser: String => Any)(sqlCommand: String, messages: String*): Unit = {
+  protected def interceptParseException(parser: String => Any)(
+    sqlCommand: String, messages: String*)(
+    errorClass: Option[String] = None): Unit = {
```

Comment:
nit: in other places, we just add the `errorClass` parameter before the `messages` parameter, e.g. `def intercept(sqlCommand: String, errorClass: Option[String], messages: String*)`, but why do we add a curry constructor here?

Reply:
The intention is to provide default value `None` for the errorClass and potential future added fields like sqlState. Especcially if new fields are added, this interface can simply be extended to 
```scala
  protected def interceptParseException(parser: String => Any)(
    sqlCommand: String, messages: String*)(
    errorClass: Option[String] = None, newField: Option[String] = None):
```
and the caller doesn't need to change its signature. For example, 
``` scala
  private def intercept(sqlCommand: String, messages: String*): Unit =
    interceptParseException(parsePlan)(sqlCommand, messages: _*)()

  private def intercept(sqlCommand: String, errorClass: Option[String], messages: String*): Unit =
    interceptParseException(parsePlan)(sqlCommand, messages: _*)(errorClass)
```
will still compile.

Reply:
WDYT? Is there another set of designs in the consideration of easy extension?